addons/mail/static/src/components/message_author_prefix/message_author_prefix.xml

```diff
@@ -2,13 +2,15 @@
 <templates xml:space="preserve">
 
     <t t-name="mail.MessageAuthorPrefix" owl="1">
-        <t t-if="message">
+        <t t-if="messageAuthorPrefixView">
             <span class="o_MessageAuthorPrefix" t-attf-class="{{ className }}" t-ref="root">
-                <t t-if="message.author and message.author === messaging.currentPartner">
-                    <i class="o_MessageAuthorPrefixIcon fa fa-mail-reply"/>You:
-                </t>
-                <t t-elif="thread and message.author !== thread.correspondent">
-                    <t t-esc="message.author.nameOrDisplayName"/>:
+                <t t-if="messageAuthorPrefixView.message">
+                    <t t-if="messageAuthorPrefixView.message.author and messageAuthorPrefixView.message.author === messaging.currentPartner">
```

Comment:
We could compute this long condition inside `messageAuthorPrefixView` inside a `isAuthorCurrentPartner` field.

I m also not sure you need to guard `messageAuthorPrefixView.message.author` since you are testing it right after. Worst case scenario is that you get `undefined === messaging.currentPartner` witch will be `false`.

Reply:
The last condition you mentioned could be true as guest.

Note that the compute could be directly on message as it only depends on message values.

Reply:
> The last condition you mentioned could be true as guest.

I don't like that this can happen. Would it be possible for `messaging.currentPartner` to have a value when it's a guest ?
Or the other way, be sure that `message.author` has a value for guest ? I think it's the already the case ?

Reply:
At the very least, this PR just introduces `MessageAuthorPrefixView/message`.
Would make another PR to introduce another field.